### PR TITLE
[refactor] Add prefix `G_` to global static variables (#3336)

### PR DESCRIPTION
**Pull request by liby**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #3336 

### Motivation

[P.NAM.09 定义全局静态变量时需加前缀G_以便和常量有所区分](https://rust-coding-guidelines.github.io/rust-coding-guidelines-zh/safe-guides/code_style/naming/P.NAM.09.html#pnam09-%E5%AE%9A%E4%B9%89%E5%85%A8%E5%B1%80%E9%9D%99%E6%80%81%E5%8F%98%E9%87%8F%E6%97%B6%E9%9C%80%E5%8A%A0%E5%89%8D%E7%BC%80g_%E4%BB%A5%E4%BE%BF%E5%92%8C%E5%B8%B8%E9%87%8F%E6%9C%89%E6%89%80%E5%8C%BA%E5%88%86)

### Undetermined part

```
vm/transaction-builder-generator/src/csharp.rs
vm/transaction-builder-generator/src/dart.rs
vm/transaction-builder-generator/src/java.rs
```

<details>
<summary>check_commit.sh output</summary>

```
Diff in /starcoin/config/src/lib.rs at line 55:
 pub use genesis_config::{
     BuiltinNetworkID, ChainNetwork, ChainNetworkID, FutureBlockParameter,
     FutureBlockParameterResolver, GenesisBlockParameter, GenesisBlockParameterConfig,
-    GenesisConfig, G_DEV_CONFIG, G_HALLEY_CONFIG, G_LATEST_GAS_SCHEDULE, G_MAIN_CONFIG, G_PROXIMA_CONFIG,
-    G_TEST_CONFIG,
+    GenesisConfig, G_DEV_CONFIG, G_HALLEY_CONFIG, G_LATEST_GAS_SCHEDULE, G_MAIN_CONFIG,
+    G_PROXIMA_CONFIG, G_TEST_CONFIG,
 };
 pub use logger_config::LoggerConfig;
 pub use metrics_config::MetricsConfig;
Diff in /starcoin/config/src/metrics_config.rs at line 179:
         self.generate_address();

         if !self.disable_metrics() {
-            let registry = Registry::new_custom(Some(G_DEFAULT_METRIC_NAMESPACE.to_string()), None)?;
+            let registry =
+                Registry::new_custom(Some(G_DEFAULT_METRIC_NAMESPACE.to_string()), None)?;
             self.registry = Some(registry);
         }
```
</details>

### CODE REVIEW

**Comment by LemonHX** [comment]
@liby please resolve conflicts

**Comment by liby** [comment]
> @liby please resolve conflicts

PTAL

**Comment by LemonHX** [comment]
@liby wait ci to pass then I will approve it

**Comment by LemonHX** [comment]
@liby fmt prob again

**Comment by liby** [comment]
> @liby fmt prob again

Already ready.

**Review by LemonHX — APPROVED** [review]

File: config/src/metrics_config.rs [hunk 1]
```diff
@@ -179,7 +179,7 @@ impl ConfigModule for MetricsConfig {
         self.generate_address();
 
         if !self.disable_metrics() {
-            let registry = Registry::new_custom(Some(DEFAULT_METRIC_NAMESPACE.to_string()), None)?;
+            let registry = Registry::new_custom(Some(G_DEFAULT_METRIC_NAMESPACE.to_string()), None)?;
```
[the remark this answers is not in the document]
Comment (liby) [reply config/src/metrics_config.rs]: It seems to be outdated.

**Review by jolestar — APPROVED** [review]

**Pull request merged**